.github/workflows/sanitizer-tests.yaml

```
@@ -101,7 +101,7 @@ jobs:
           target_directory: src
           googletest: true
           discovery_server: true
-          discovery_server_branch: ${{ github.event.inputs.discovery_server_branch || env.DEFAULT_DISCOVERY_SERVER_BRANCH }}
+          discovery_server_branch: ${{ github.base_ref || github.head_ref || github.event.inputs.discovery_server_branch || env.DEFAULT_DISCOVERY_SERVER_BRANCH }}
```

I think we should prioritize source branch (head) over target branch (base). Furthermore, as far as I can tell, when manually run, the specified branch does not have priority over neither head or base. I don't know if we can detect that a workflow has been manually run and the branch as been explicitly set. Else, we could make the option mandatory in that case, thus avoiding the default value, and always prioritize the option.

```suggestion
          discovery_server_branch: ${{ github.head_ref || github.base_ref || github.event.inputs.discovery_server_branch || env.DEFAULT_DISCOVERY_SERVER_BRANCH }}
```

---

`github.base_ref` and `github.head_ref` only exists if the workflow is running from a PR. The same happens with `github.event.inputs.discovery_server_branch` but for workflow dispatch. Knowing that, we can guarantee the the workflow is running from a PR, a workflow dispatch or another event (push on main branch). In any case, I've just seen that according to this, the `github.base_ref` will never be checked so we need to find a way to pass both to the action.

---

Addressed in 58c7943